### DISCUSSION TITLE
why does my yt-dlp not delete these temp files?

### DISCUSSION THREAD

**Discussion by skotarek** [mpv-player/mpv#14461] [Q&A]
![temp](https://github.com/mpv-player/mpv/assets/76743591/4256c7d3-c4cf-42da-a150-21a44d00c757)

as far as I understand the files in the %temp% folder generated by yt-dlp.exe are supposed to automatically delete themselves after closing mpv or something similar, however only some of them do and most remain on my pc taking up gigabytes of space

I think this happens most often when I'm running a playlist as I haven't noticed them not being deleted when running a single video but I'm not entirely sure about that

what can I possibly do here? what could be the cause and what is the solution? I'm looking for a more in-depth explanation rather than just a workaround like creating a batch file to remove these files and running it on startup

**Comment by avih** [comment]
This is a yt-dlp issue, and not an mpv issue.

I'm guessing that the single exe yt-dlp.exe needs to extract itself, and it does it to %TEMP%. You should ask them why it doesn't clean up completely.

As an alternative, you can download the yt-dlp windows zip package instead of the single-executable, extract it someplace, and then tell mpv where it is, e.g. at your mpv.conf add this line (with the path adjusted to your system):
```
script-opts-add=ytdl_hook-ytdl_path=c:/yt-dl/yt-dlp.exe
```

Then it's already extracted and should not create those directories at the temp dir.

**Reply by skotarek** [reply]
> I'm guessing that the single exe yt-dlp.exe needs to extract itself, and it does it to %TEMP%. You should ask them why it doesn't clean up completely.

okay so I noticed that these _MEI files get deleted only if I'm running a playlist and skipping to the next video before current one finishes its startup load - at that point I think what happens is that the cleaning function breaks itself
I also asked about this in a yt-dlp issue and was told that this is something yt-dlp devs can't help with as it's related to mpv or possibly pyinstaller so here I am again

**Comment by avih** [comment]
I don't see how it can be related to mpv.

mpv executes yt-dlp.exe (or whatever other name it has) and captures its output. It doesn't care and doesn't control how yt-dlp.exe runs, and it doesn't know and doesn't control whether or not it extracts itself or does any other thing.

This is completely outside the control of mpv.

**Reply by skotarek** [reply]
then do you think it's something that might be related to pyinstaller?

**Comment by avih** [comment]
I don't know what "pyinstaller" refers to.

I'm guessing it's related to whatever thing which is used to create a single windows executable from a python project with several files.

This thing probably handles the packaging of a python project into a single windows executable, and then handles the startup/extraction/run/cleanup of that single exe.

I don't know which tools are used to achieve that.
